### DISCUSSION TITLE
Flyout Navigation and General Navigation is slower compared to Xamarin forms in MAUI

### DISCUSSION THREAD

**Discussion by dharamhbtik** [PrismLibrary/Prism#3286] [.NET MAUI]
We recently migrated from Xamarin Forms to MAUI with .NET 8. We have used PRISM extensively in Xamarin Forms and MAUI, but we are using PRISM only for navigation and registered Views and View Models only in the PRISM IoC through PRISM startup.

Main Navigation through NavigateAsync method is also slower than Xamarin Forms and the Flyout Menu is also slower.

Is there any tweak we need to do in the MAUI to make it faster or at least similar to Xamarin Forms' performance? 

The rest of the dependencies we are injecting through Microsoft inbuild dependencies. 

```
 .RegisterTypes()
 .UsePrism(Rules.Default.WithMicrosoftDependencyInjectionRules(), PrismStartup.Configure)
```

In the above code, registerTypes we are registering the other dependencies, and in PrismStartup.Configure we are registering the Views and ViewModels 

For Menu Navigation 

`await _navigationService.NavigateAsync($"/{nameof(MenuPage)}/{nameof(NavigationPage)}/{nameof(HomePage)}");`

for normal navigation 
` await _navigationService.NavigateAsync($"/{nameof(NavigationPage)}/{nameof(SettingsPage)}?{HomePageViewModel.InitData}=true");`

  Any suggestions for performance improvement?

**Comment by HelenMamalaki** [comment]
Does your navigation slowness make the app appear slow to start up? or slow to use as you navigate around within the app? Just wondering if it's similar to my issue #3312 What platform has the slowness or is it all platforms?
